nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -279,9 +280,22 @@ Costmap2DROS::on_activate(const rclcpp_lifecycle::State & /*state*/)
   // Create a thread to handle updating the map
   stopped_ = true;  // to active plugins
   stop_updates_ = false;
-  map_update_thread_shutdown_ = false;
-  map_update_thread_ = std::make_unique<std::thread>(
-    std::bind(&Costmap2DROS::mapUpdateLoop, this, map_update_frequency_));
+
+  if (!map_update_timer_) {
+    RCLCPP_DEBUG(get_logger(), "mapUpdateLoop frequency: %lf", map_update_frequency_);
+    if (map_update_frequency_ > 0.) {
+      map_update_timer_ = rclcpp::create_timer(
+        this, get_clock(), rclcpp::Duration::from_seconds(1 / map_update_frequency_),
+        [this]() {mapUpdateLoop();});
+    } else {
+      RCLCPP_WARN(
+        get_logger(),
+        "Costmap won't be periodically updated as frequency %lf should be greater than zero.",
+        map_update_frequency_);
+    }
+  } else {
+    map_update_timer_->reset();
```

Comment:
if `map_update_timer_` its a nullptr, why reset it?

Reply:
I think that may actually be dereferencing a nullptr

Reply:
If `map_update_timer_` is a nullptr, we are trying to create a new timer, else - resetting existing timer:
```
if (!map_update_timer_) {
  // map_update_timer_ == nullptr
  try to create a new map_update_timer_ object
} else {
  // map_update_timer_ != nullptr
  // which means that timer is already created. We just need to reset it:
  map_update_timer_->reset();
}
```
I looks like there are no cases where we could dereference a null pointer.